packages/components/src/RecurringSelect/components/MonthlyDayOfWeekSelect/MonthlyDayOfWeekSelect.css

```diff
@@ -0,0 +1,37 @@
+.parentContainer {
+  /* useless, only here to avoid linting error */
+  display: block;
+}
+
+.container {
+  display: flex;
+  margin-bottom: calc(-1 * var(--border-base));
+}
+
+.parentContainer > div:nth-of-type(1) > div:nth-of-type(2) div {
```

Comment:
I'm not proud of everything below this

it's the only way to have the individual WeeklySelects *not* have their standard border radius that we do indeed want for individual cases

when they're stacked as 4, they each need unique border radius rules

this pairs with the `parentContainer` which is the only element in this component that can effectively keep count of how many of the WeeklySelects we have

we can't target their class names since those get generated but within the scope of the other component

Reply:
@zek is there a way the component could be smart enough to give classnames to first and last row containers? like



```
.MonthlyDayOfWeekSelect--selector {
  border-radius: 0; (or I guess we wouldn't even need this)
}

.MonthlyDayOfWeekSelect--container.MonthlyDayOfWeekSelect--first .MonthlyDayOfWeekSelect--selector {
  border-top-right-radius: var(--radius-large);
  border-top-left-radius: var(--radius-large);
}

.MonthlyDayOfWeekSelect--container.MonthlyDayOfWeekSelect--last  .MonthlyDayOfWeekSelect--selector {
  border-bottom-right-radius: var(--radius-large);
  border-bottom-left-radius: var(--radius-large);
}
```

Or something like that?

Reply:
Basically just wondering if there are any opportunities to get more classnames in there so it's a little less brittle

Reply:
I get what your idea is, and it almost works but it ends up not applying I believe due to the way the css definitions get exported, imported and ultimately used

even with specific class names assigned to the 1-4 instances, what happens is still comparable to what we have here

we can't write a class in the template, or React component of say `MonthlyDayOfWeekSelectFirst` and have that be applied. what happens is that the markup will get `.MonthlyDayOfWeekSelectFirst` as a class, but that's not what the exported css class is.

instead what happens is that React, or whoever really, adds a random affix to ensure uniqueness/scope. this means the styles never match and don't get applied

in order for the styles to apply, you must have a `MonthlyDayOfWeekSelectFirst` definition in your `css` which then gets added to the generated `css.d.ts` file for it

then you must actually reference that in your component via `styles.MonthlyDayOfWeekSelectFirst` , meaning the `--first` syntax becomes pretty awkward. I'm not even sure if you can do that, even with a `['MonthlyDayOfWeekSelect--first']` syntax, but I didn't try admittedly

so I can get that to work, but what happens is that we're back at square 1. arguably worse. since we don't actually care about `MonthlyDayOfWeekSelectFirst`, because it's the immediate child div that we want to apply styles to, that means we have no rule of say

```
.MonthlyDayOfWeekSelectFirst { content: "pls hlp"; } 
```

rather what we have, and need is 

```
.MonthlyDayOfWeekSelectFirst div { border-radius: 5px; }
```

 but we can't apply that second rule via `styles.MonthlyDayOfWeekSelectFirst div` that's not a thing. we can only apply `styles.MonthlyDayOfWeekSelectFirst`. this means we have to have that empty definition in the css, but the linter doesn't like that. so now we need 4 empty definitions, plus 4 custom border radii selectors on the child div, which is still pretty fragile imo, and adds even more clutter

TL;DR - we can do it but I don't know if it's actually any better :/

Reply:
https://media.giphy.com/media/xUPGcIm3J0teFwk1UI/giphy.gif

Reply:
after chatting with Darryl, it might not be as bad as I thought with the empty rules. that was my IDE being a slowpoke

I'll see if I still have the work I tried here stashed